File: frontend/src/api/post.ts
import { http } from '@/utils/request';
import type { Comment, PageData, Post, Topic } from '@/types';

export const postApi = {
  list: (params?: { page?: number; page_size?: number }) =>
    http.get<PageData<Post>>('/posts', params),
  get: (id: string) => http.get<Post>(`/posts/${id}`),
  create: (data: Partial<Post>) => http.post<Post>('/posts', data),
  listByAuthor: (userId: string, params?: { page?: number; page_size?: number }) =>
    http.get<PageData<Post>>(`/users/${userId}/posts`, params),
  listByTopic: (name: string, params?: { page?: number; page_size?: number }) =>
    http.get<PageData<Post>>(`/topics/${encodeURIComponent(name)}/posts`, params),
  listComments: (postId: string, params?: { page?: number; page_size?: number }) =>
    http.get<PageData<Comment>>(`/posts/${postId}/comments`, params),
  addComment: (postId: string, content: string) =>
    http.post<Comment>(`/posts/${postId}/comments`, { content }),
  deleteComment: (postId: string, commentId: string) =>
    http.delete<{ message: string }>(`/posts/${postId}/comments/${commentId}`),
  interact: (postId: string, type: 'like' | 'favorite' | 'forward') =>
    http.post<{ active: boolean; type: string }>(`/posts/${postId}/interact`, { type }),
  topics: (params?: { page?: number; page_size?: number }) =>
    http.get<PageData<Topic>>('/topics', params),
};

File: backend/internal/router/post.go
package router

import (
	"github.com/gin-gonic/gin"

	"github.com/petsocial/petsocial/internal/handler"
)

// RegisterPostPublic 动态公开路由。
func RegisterPostPublic(r *gin.RouterGroup, h *handler.PostHandler) {
	r.GET("/posts", h.List)
	r.GET("/posts/:id", h.Get)
	r.GET("/posts/:id/comments", h.ListComments)
	r.GET("/topics", h.ListTopics)
	r.GET("/topics/:name/posts", h.ListByTopic)
}

// RegisterPostAuth 动态登录路由。
func RegisterPostAuth(r *gin.RouterGroup, h *handler.PostHandler) {
	r.POST("/posts", h.Create)
	r.POST("/posts/:id/comments", h.AddComment)
	r.DELETE("/posts/:id/comments/:comment_id", h.DeleteComment)
	r.POST("/posts/:id/interact", h.Interact)
	r.GET("/users/:id/posts", h.ListByAuthor)
}
